org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/job/DisplayProjectRuleDescriptionJob.java

```diff
@@ -19,42 +19,43 @@
  */
 package org.sonarlint.eclipse.ui.internal.job;
 
-import java.util.concurrent.TimeUnit;
 import org.eclipse.core.runtime.CoreException;
 import org.eclipse.core.runtime.IProgressMonitor;
 import org.eclipse.core.runtime.IStatus;
 import org.eclipse.core.runtime.Status;
 import org.eclipse.swt.widgets.Display;
 import org.sonarlint.eclipse.core.SonarLintLogger;
-import org.sonarlint.eclipse.core.internal.engine.connected.ResolvedBinding;
+import org.sonarlint.eclipse.core.internal.backend.SonarLintBackendService;
 import org.sonarlint.eclipse.core.internal.jobs.AbstractSonarProjectJob;
 import org.sonarlint.eclipse.core.resource.ISonarLintProject;
 import org.sonarlint.eclipse.ui.internal.util.SonarLintRuleBrowser;
 
-public class AsyncDisplayRuleDescriptionJob extends AbstractSonarProjectJob {
-  private final ResolvedBinding binding;
+/** Update "web browser" view for the project rule description (maybe context based on connection) in a separate thread */
+public class DisplayProjectRuleDescriptionJob extends AbstractSonarProjectJob {
+  private final ISonarLintProject project;
   private final String ruleKey;
+  private final String contextKey;
   private final SonarLintRuleBrowser browser;
 
-  public AsyncDisplayRuleDescriptionJob(ISonarLintProject project, ResolvedBinding binding, String ruleKey, SonarLintRuleBrowser browser) {
+  public DisplayProjectRuleDescriptionJob(ISonarLintProject project, String ruleKey, String contextKey, SonarLintRuleBrowser browser) {
```

Comment:
You can mark `contextKey` as Nullable

Reply:
Done, also in `SonarLintBackendService.getEffectiveRuleDetails`.